skada/_ot.py

```diff
@@ -728,7 +728,17 @@ def predict(self, X, sample_domain=None, *, sample_weight=None):
             )
         return self.estimator_.predict(X)
 
-    def score(self, X, y, sample_domain=None, *, sample_weight=None):
+    def predict_proba(self, X, sample_domain=None, *, sample_weight=None):
+        """Predict using the model"""
+        check_is_fitted(self)
+        if sample_domain is not None and np.any(sample_domain >= 0):
+            warnings.warn(
+                "Source domain detected. Predictor is trained on target"
+                "and prediction might be biased."
+            )
+        return self.estimator_.predict_proba(X)
```

Comment:
This function should not be available whene teh abse estimator does not provide predict_proba. use the `available_if` 
https://scikit-learn.org/stable/modules/generated/sklearn.utils.metaestimators.available_if.html

Look how they did for SVC:
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/svm/_base.py#L821